### ISSUE TITLE
question: multi-lane use meso

### ISSUE THREAD

**Issue by ambuehll** [eclipse/sumo#5709]
I am currently trying to figure out why the vehicles are not using multiple lanes on highways. To my understanding, the lanes in the meso-model are only used, if there is a queue on a lane at all. 
Somehow from the GUI screenshot below, I have the impression this is not done - or the vehicles are already pre-sorted for the exit further downstream. 
The problem with this, however, that unnecessary gridlocks are created upstream even though the highway can store many more vehicles. Any suggestions highly appreciated. 

![image](https://user-images.githubusercontent.com/48930860/58880009-35c98280-86d7-11e9-9d29-738ddd3de314.png)

reproducible with 
sumo v1_2_0+0639-3dcaea6686
[files](https://drive.google.com/open?id=15j5KzyHJ_Yo0a9_a7WoRfXeuAy9gYsRN)
with scale value="1"
at junction:30033686

**Comment by namdre** [comment]
meso only accounts for multiple lanes when the lanes have different successor edges.
On a multi-lane highway where each edge typically has only 1 successor, the headway values are scaled according to the lane number (effectively scaling the maximum flow). Thus you will see all vehicles rendered on the rightmost lane but this by itself will not cause jamming or cause invalid maximum density. (you can right-click on the segment to check occupancy and jam threshold).

If you can provide a smaller input sample (ideally with only the relevant part of the motorway) then I can take a look for unexpected behavior.

**Comment by ambuehll** [comment]
Ah, I didn't know about the rendering. Makes sense now. Many thanks!
If you're still interested, I have updated the link to include files with "_crop" that contain a tiny part of the network only. It seems as if it were a general grid-lock problem due to (too) high demand. One thing that I see frequently, is relatively high flows (>100veh/h-lane) with very low speeds (e.g. 0.2 m/s). This is obviously an artefact of the way the meso model is set up, however, it would be nice if the likelihood of such cases appearing could be reduced.

**Comment by namdre** [comment]
- you have set quite high values for the tau parameters. These denote net-time gaps (front-bumper to back-bumper) and the way you set them looks more like gross time gaps.
- the default jamming threshold still dates to the time where meso vehicles all drove at the same speed by default. When vehicles follow a speed distribution it is more apppropriate to relax the jamming treshold. This can be done by setting
  --meso-jam-threshold -0.5 (the closer to 0 the stronger the relaxiation whereas -1 is the default).
See https://sumo.dlr.de/wiki/Simulation/Meso#Longitudinal_Model

**Comment by ambuehll** [comment]
Thanks for the clarification! 

- You are right, these were brutto headways (which I defined after reading Eisfeldt's thesis). May I suggest that this information is also added to the wiki. The current definition might be confusing as, headways are usually defined between (e.g.) front bumper of two consecutive vehicles (brutto gaps) (check e.g. Daganzo 2008). 

- The standard deviation in my speed distribution is very low (0.03) in order to avoid stochastic speeds. I will play around with the meso-jam-treshold. Any benchmarks/ideas?

**Comment by namdre** [comment]
1) I already modified the wiki to make this more clear and also explain the reason.

2) Eissfeldt used an explicit jam threshold parameter (which you can still do by giving a positive value). 
This creates issues when edges have wildly varying speed limits (typically in large networks) due to effect on free-flow density.
The current default of -1 computes the value so that vehicles driving at free flow speed never count as jammed. However, one could argue that this condition is far too harsh because in reality a motorway might still see high flows even if the speed drops to 70% of the speed limit. (which would suggest a value of -0.7)

Eissfeldt defines a car as jammed when it has less than 50% of the speed limit which would suggest a value of -0.5.

The best way would be to calibrate the jam threshold value with the help of real-world traffic data for your region.

**Comment by ambuehll** [comment]
OK. Thanks for the very valuable feedback! I investigated the influence of 2: Based on your explanation, I am expecting a shift in the FD as follows: base shape is an inverse lambda FD with a max flow roughly at 1/tau_ff (exaggerated in the figure below), shift to the right with a plateau (at 1/tau_jf or 1/tau_ff (?)). 
![schematic FD](https://user-images.githubusercontent.com/48930860/59050684-3acf3300-888b-11e9-9b54-e354e703b1d1.png)

This is more or less supported by the following two links evaluated in the [usual network](https://drive.google.com/drive/folders/1DFXy8Skc-5ZQ7rZ4SJBfDuAj89rUkk0Y?usp=sharing) (6min flow and density averages)
![FD of two links](https://user-images.githubusercontent.com/48930860/59050852-b4672100-888b-11e9-81c6-cf27b67ae194.png)

So, in other words, the meso-jam-treshold defines the form of the FD.

**Comment by namdre** [comment]
jam-threshold is definitely a strong influence on the shape of the FD because it defines the point where the flow break-down occurs.
